sdcflows/transform.py

```diff
@@ -49,174 +227,291 @@ class B0FieldTransform:
 
     coeffs = attr.ib(default=None)
     """B-Spline coefficients (one value per control point)."""
-    xfm = attr.ib(default=None, on_setattr=_clear_mapped)
-    """A rigid-body transform to prepend to the unwarping displacements field."""
     mapped = attr.ib(default=None, init=False)
     """
     A cache of the interpolated field in Hz (i.e., the fieldmap *mapped* on to the
     target image we want to correct).
     """
 
-    def fit(self, spatialimage):
+    def fit(
+        self,
+        target_reference: nb.spatialimages.SpatialImage,
+        fmap2data_xfm: np.ndarray = None,
+        approx: bool = True,
+    ) -> bool:
         r"""
         Generate the interpolation matrix (and the VSM with it).
 
         Implements Eq. :math:`\eqref{eq:1}`, interpolating :math:`f(\mathbf{s})`
         for all voxels in the target-image's extent.
 
+        Parameters
+        ----------
+        target_reference : :obj:`~nibabel.spatialimages.SpatialImage`
+            The image object containing a reference grid (same as that of the data
+            to be resampled). If a 4D dataset is provided, then the fourth dimension
+            will be dropped.
+        fmap2data_xfm : :obj:`numpy.ndarray`
+            Transform that maps coordinates on the `target_reference` onto the
+            fieldmap reference (that is, the linear transform through which the fieldmap can
+            be resampled in register with the `target_reference`).
+            In other words, `fmap2data_xfm` is the result of calling a registration tool
+            such as ANTs configured for a linear transform with at most 12 degrees of freedom
+            and called with the image carrying `target_affine` as reference and the fieldmap
+            reference as moving.
+            The result of such a registration framework is an affine (our `fmap2data_xfm` here)
+            that maps coordinates in reference (target) RAS onto the fieldmap RAS.
+        approx : :obj:`bool`
+            If ``True``, do not reconstruct the B-Spline field directly on the target
+            (which will likely not be aligned with the fieldmap's grid), but rather use
+            the fieldmap's grid and then use just regular interpolation.
+
         Returns
         -------
         updated : :obj:`bool`
             ``True`` if the internal field representation was fit,
             ``False`` if cache was valid and will be reused.
 
         """
+        import pdb; pdb.set_trace()
```

Comment:
This is causing CircleCI to hang indefinitely. We should probably remove `-it` from `docker run` commands to avoid this.

```suggestion
```